### ISSUE TITLE
Needs different merge strategy

### ISSUE THREAD

**Issue by braco** [kota65535/openapi-merger#31]
Common.yml
```
properties:
  things:
    type: object
    properties:
        foo: ...
        bar: ...
```

Extended.yml
```
$include: "./Common.yml"
description: ...
properties:
  types: ...
  tags: ...
  things:
    properties:
        baz: ...
        boo: ...
 ```

The intention here is that Common.yml has some `things` that are merged with Extended.yml. However, the merge strategy isn't deep, so Extended overwrites everything in Common, yielding only baz and boo.

edit: this seems like it may be a bug since two includes:
```
$include#1: ...
$include#2: ...
```
seem to merge deeply, but not include and an inline object.

**Comment by jameskhamil** [comment]
This just burn me. @kota65535 what do you think of this feature?

I try to $include a file and then overwrite one nested property, it erased the whole tree. Needs a deep merge.

The behavior is different with two includes and inline like @braco says!

**Comment by braco** [comment]
@jameskhamil yes I would consider this package abandoned. I changed the behavior locally, but based on getting no responses here I didn't bother setting up a PR.

The merge strategy in this code is naive, you can just switch it out for one of the hundreds of merge packages on npm:
https://github.com/kota65535/openapi-merger/blob/main/src/merger.js#L69

**Comment by kota65535** [comment]
@jameskhamil @braco Hey guys, sorry for the long time inconvenience,  I'm working on it.

**Comment by kota65535** [comment]
@jameskhamil @braco I've released 0.5.3. Please take a look.
